asyncio_pool/base_pool.py

```diff
@@ -106,7 +106,7 @@ async def _wrap(self, coro, future, cb=None, ctx=None):
         res, exc, tb = None, None, None
         try:
             res = await coro
-        except Exception as _exc:
+        except BaseException as _exc:
```

Comment:
This catch now includes stuff that, in my opinion, should never be catched in library code, e.g. `SystemExit` and `KeyboardInterrupt`.

https://docs.python.org/3/library/exceptions.html#exception-hierarchy

Reply:
Yes but no. The job of these future wrappers is to specifically catch *all* exceptions raised in the wrapped future and then set them as the future's exception. You're not swallowing the exception, you are making it available to be fetched on the call-site in the first palce. They would be swallowed if you didn't, since the exception would bubble up in the task and then never be retrieved.